control-plane/go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
 	github.com/hashicorp/consul-k8s/control-plane/cni v0.0.0-20220831174802-b8af65262de8
 	github.com/hashicorp/consul-server-connection-manager v0.1.0
-	github.com/hashicorp/consul/api v1.18.0
+	github.com/hashicorp/consul/api v1.10.1-0.20221220195433-629878a6879c
```

Comment:
we should look into this sometime in the future because it is awfully confusing that the API module version is `1.10.*`.

Reply:
Hard agree that it's messy. If I understand how the [pseudo-version](https://go.dev/ref/mod#pseudo-versions) is created, it's because the last semver tag on Consul `main` for `api` module was `api/1.10.0`.  You can tell it's a holiday week because I made a shell snippet to verify:
```bash
git tag -l 'api/*' | while read tag
do
    echo Tag is $tag
    commit=$(git rev-list -n 1 $tag)
    echo Commit is $commit
    if git merge-base --is-ancestor $commit HEAD; then
        echo "Yes"
    else
        echo "No"
    fi
    echo ""
done
```

All this to say, I'm not sure I have no idea how to fix it 😬 .